src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -462,8 +468,14 @@ def _compile_hermetic(self, jvm_options, ctx, classes_dir, jar_file,
       (self.post_compile_extra_resources_digest(ctx), argfile_snapshot.directory_digest)
     )
 
+    env = {}
+    agent_lib_dir = self.get_options().native_image_agent_library_dir
+    if agent_lib_dir:
+      env[Platform.current.runtime_lib_path_env_var] = agent_lib_dir
```

Comment:
Rather than leaking a probably-absolute path into the env here, should we capture the file as a snapshot (probably modifying the `argfile_snapshot` call above, so we can do the capturing in parallel), merge that into the `merged_input_digest`, and use the path within that as the env var?

Reply:
The reason why we wouldn't immediately be able to do this is that typically these env vars require absolute paths, and there's currently not a way to get an absolutified version of a relative path within the hermetic execution sandbox without shelling out to `sh` -- see the kind of manipulation that is currently used for the subprocess argv in #6893: https://github.com/pantsbuild/pants/blob/24727b2273b4bd6436bb3411dce04f9a8e64c9b0/src/python/pants/backend/jvm/subsystems/graal.py#L236-L251

Reply:
Could you use:
```
java -help
...
    -agentpath:<pathname>[=<options>]
                  load native agent library by full pathname
```
...to load the agent instead?

----

If not, this should probably use `subprocess`. The need to not violate the `hermetic` sandbox is getting very real.

Reply:
...it looks like that works great! We can avoid all of this now! Thank you!

Reply:
I think I'd be fine with introducing a helper function / `@rule` something along the lines of:
```python
def execute_process_with_env_vars_interpolated(execute_process_request, env_to_command):
  with temporary_directory() as tempdir:
    with open(os.path.join(tempdir, "script"), "w") as f:
      lines = ["{key}={value}".format(key=key, value=value) for key, value in sorted(env_to_command.items())]
      lines.append("exec " + shlex(execute_process_request.argv))
      f.write("\n".join(lines))
    captured_directory_digest = capture(tempdir, "script")
    merged_directory_digest = merge(captured_directory_digest, execute_process_request.directory_digest)
  return execute_process_request.with(argv="./script", directory_digest=merged_directory_digest)
````

which doesn't seem super crazy, and would let us actually execute this all hermetically by calling `execute_process_with_env_vars_interpolated(epr, {"PWD": "/bin/pwd"})`

Reply:
I would be a huge fan of this!! Broke out into #8076!